client/plugin/scripts/popup.js

```diff
@@ -29,6 +29,30 @@ function searchAPI(keywords, resultsCount) {
     });
 }
 
+// function getFaviconUrl(url) {
```

Comment:
This is great, will try it out separately and if all looks good, we can add a PR for this separately.

Reply:
This won't work for all urls, this was made keeping in mind Wikimedia primarily, we need to write an intelligent way to figure out favicons, this is just a rudimentary hack I tried out and wanted opinions